modules/shapefile/src/lib/parse-geometry.js

```diff
@@ -143,9 +143,19 @@ function parsePoly(view, offset, dim, type) {
 
   const positions = concatPositions(xyPositions, mPositions, zPositions);
 
+  if (type === 'LineString') {
```

Comment:
Maybe a short one-line  comment explaining why this is the only case that needs special treatment

Reply:
The only valid Polyline input types are Linestrings and Polygons (they're stored the same way). Points are handled in a separate function